### PR TITLE
Add My GitHub PRs provider with review status tracking

### LINKED ISSUE #276: Automatically track PRs and surface their status
Track PRs I've created and surface their current status as work items. Statuses to track:

- **Waiting on reviews** — PR is open, no reviews yet
- **Review received** — reviewer has commented or requested changes
- **Ready to merge** — approved and CI green
- **Merged** — PR has been merged

This would let me see at a glance where all my open PRs stand without leaving DevDocket.

### PR DESCRIPTION
**Pull request by mthalman**
## Summary

Adds a new **My GitHub PRs** provider that automatically discovers open pull requests authored by the current user and surfaces their review/CI status within DevDocket.

### Status values

- **Draft** — PR is in draft mode
- **Waiting on reviews** — No reviews received yet
- **Review received** — Reviewers have commented (but no approval/rejection decisions)
- **Changes requested** — At least one reviewer's latest decision is "changes requested"
- **Approved** — All review decisions are approvals, but CI/branch protection isn't green yet
- **Ready to merge** — Approved and `mergeable_state` is clean (CI green, no conflicts)

### How it works

1. Uses the GitHub Search API (`type:pr author:@me state:open`) to discover the user's open PRs
2. For each PR, fetches details and reviews via the REST API using concurrent workers (max 3)
3. Computes status from the latest review decision per reviewer and the PR's `mergeable_state`
4. Emits discovered items with status shown via `DiscoveredItem.state`

The provider reuses the existing `devdocketGithub.repos` configuration — when repos are specified, only those repos are searched; otherwise all repos are included.

### Testing

24 new tests covering:
- Provider discovery flow (auth, search, enrichment)
- All 6 status determination paths via pure `determinePrStatus()` unit tests
- Edge cases: draft priority, multiple reviewers, dismissed reviews, missing user IDs, concurrent worker mock ordering

Closes #276

### CODE REVIEW

File: packages/github/src/test/githubMyPrsProvider.test.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+      { draft: false },
+      [
+        { user: { id: 1 }, state: 'CHANGES_REQUESTED', submitted_at: '2025-01-01T00:00:00Z' },
+        { user: { id: 1 }, state: 'DISMISSED', submitted_at: '2025-01-02T00:00:00Z' },
+      ],
+    );
+    // DISMISSED doesn't override — the previous CHANGES_REQUESTED still stands
+    // but since DISMISSED is filtered out, the CHANGES_REQUESTED is the latest decision
+    expect(status).toBe('Changes requested');
+  });
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/githubMyPrsProvider.test.ts]: Fixed in d749170 — DISMISSED now neutralizes prior decisions and test updated accordingly.

File: packages/github/src/githubMyPrsProvider.ts [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      }
+
+      const existing = latestByReviewer.get(userId);
+      if (
+        !existing ||
+        (review.submitted_at && existing.submitted_at && review.submitted_at > existing.submitted_at)
+      ) {
+        latestByReviewer.set(userId, review);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts:271]: Fixed in d749170 — DISMISSED now tracked as a decision state and filtered from active decisions after building the per-reviewer map.

File: packages/github/src/githubMyPrsProvider.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+      }
+      if (review.state !== 'APPROVED' && review.state !== 'CHANGES_REQUESTED') {
+        continue;
+      }
+
+      const existing = latestByReviewer.get(userId);
+      if (
+        !existing ||
+        (review.submitted_at && existing.submitted_at && review.submitted_at > existing.submitted_at)
+      ) {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in d749170 — submitted_at now defaults to empty string when missing, ensuring later reviews with timestamps can always override earlier ones without timestamps.

File: packages/github/src/githubMyPrsProvider.ts [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+    // Fetch reviews
+    const reviewsUrl = `${pr.pull_request!.url}/reviews`;
+    const reviewsResponse = await fetch(reviewsUrl, { headers });
+    let reviews: PrReview[] = [];
+    if (reviewsResponse.ok) {
+      reviews = (await reviewsResponse.json()) as PrReview[];
+    } else {
+      logger.debug(`Failed to fetch reviews for ${pr.html_url}: ${reviewsResponse.status}`);
+    }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in f87b90c — reviews fetch failure now returns undefined, falling back to Open status.

File: packages/github/src/githubMyPrsProvider.ts [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+    const latestByReviewer = new Map<number, PrReview>();
+    for (const review of reviews) {
+      const userId = review.user?.id;
+      if (userId === undefined || !review.state) {
+        continue;
+      }
+      if (review.state !== 'APPROVED' && review.state !== 'CHANGES_REQUESTED' && review.state !== 'DISMISSED') {
+        continue;
+      }
+
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts:264]: This matches GitHub's own review model. A COMMENTED review is "leave comments without a decision" — it doesn't clear a prior APPROVED or CHANGES_REQUESTED. On github.com, if reviewer A requests changes and then submits a COMMENTED review, the PR still shows "Changes requested" from that reviewer. Only APPROVED or DISMISSED clears the prior decision. Keeping this behavior ensures our status matches what users see on GitHub.

File: packages/github/src/githubMyPrsProvider.ts [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    });
+
+    this._onDidDiscoverItems.fire(items);
+
+    if (failures.length > 0) {
+      const message = failures.length === 1
+        ? `Failed to fetch authored PRs from ${failures[0]}`
+        : `Failed to fetch authored PRs from ${failures.length} repositories`;
+      if (isUserTriggered) {
+        vscode.window.showWarningMessage(`DevDocket GitHub: ${message}`);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in 561cc44

File: packages/github/src/test/githubMyPrsProvider.test.ts [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+function mockFailedResponse(status = 500) {
+  return { ok: false, status, statusText: 'Internal Server Error' };
+}
+
+describe('GitHubMyPrsProvider', () => {
+  let provider: GitHubMyPrsProvider;
+  let mockChannel: { appendLine: ReturnType<typeof vi.fn> };
+
+  beforeEach(() => {
+    vi.clearAllMocks();
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/githubMyPrsProvider.test.ts]: Fixed in 15cdc87

File: packages/github/src/githubMyPrsProvider.ts [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+/**
+ * DevDocket provider that discovers GitHub pull requests authored by the
+ * current user and surfaces their review/CI status.
+ *
+ * Uses the GitHub Search API (`author:@me`) to find open PRs, then enriches
+ * each with review decisions and mergeable state via the PR detail and
+ * reviews REST endpoints.
+ *
+ * Status values: Draft, Waiting on reviews, Review received,
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in b6b49c5
